modules/db/postgres.py

```diff
@@ -19,24 +21,59 @@ def __init__(self):
     def __del__(self):
         self.session.close()
 
+    def select_id_from_table(self, table_name: str, id_name: str, elem_name: str, elem_value: str):
+        query = sql.SQL("SELECT {id} from {table} WHERE {name} = %s").format(
+            id=sql.Identifier(id_name), 
+            table=sql.Identifier(table_name), 
+            name=sql.Identifier(elem_name))
+        self.cursor.execute(query, [str(elem_value)])
+
+        return self.cursor.fetchone()
+
+    def insert_new_element_to_table(self, table_name: str, id_name: str, elem_name: str, elem_value: str, elem_id: str):
+        query = sql.SQL("INSERT INTO {table} ({fields}) VALUES (%s, %s)").format(
+            table=sql.Identifier(table_name), 
+            fields=sql.SQL(',').join([
+                sql.Identifier(id_name),
+                sql.Identifier(elem_name)]))
+
+        self.cursor.execute(query, [elem_id, str(elem_value)])
+    
+
+    def get_element_id(self, table_name: str, id_name: str, elem_name: str, elem_value: str):
+        elem_id = self.select_id_from_table(table_name, id_name, elem_name, elem_value)
+
+        if (elem_id is not None):
+            return str(elem_id[0])
+
+        elem_id = str(uuid.uuid4())
+        self.insert_new_element_to_table(table_name, id_name, elem_name, elem_value, elem_id)
+        return elem_id
+
+
+    def set_insert_query(self, record):
+        brand_id = self.get_element_id("car_brands", "brand_id", "brand", record["car_brand"])
```

Comment:
why separate calls to fetch ids instead of performing one sql query using subqueries?
(select id from car_classes where ...)

Reply:
insert changed